lib/soft_timer_delay.h

```diff
@@ -77,11 +77,8 @@
 
 #include <stdint.h>
 
-enum
-{
-    /** Standard timer frequency, corresponding to a 32kHz clock */
-    SOFT_TIMER_FREQUENCY = 32768,
-};
+/** Standard timer frequency, corresponding to a 32kHz clock */
+#define SOFT_TIMER_FREQUENCY 32768
```

Comment:
The common practice is to make every define safe, i. e. with parentheses if
it is more complex that a literal. The parentheses here are useful only in
cases where SOFT_TIMER_FREQUENCY itself is not safe (e. g. defined to 3+2
with no parentheses). In Contiki we assume the defines are safe and we
would generally not use parentheses here (this was not my code ;))
Le 16 févr. 2016 17:34, "Gaëtan Harter" <notifications@github.com> a écrit :

> In lib/soft_timer_delay.h
> <https://github.com/iot-lab/openlab/pull/5#discussion_r53036521>:
>
> > @@ -77,11 +77,8 @@
> >
> >  #include <stdint.h>
> >
> > -enum
> > -{
> > -    /** Standard timer frequency, corresponding to a 32kHz clock */
> > -    SOFT_TIMER_FREQUENCY = 32768,
> > -};
> > +/** Standard timer frequency, corresponding to a 32kHz clock */
> > +#define SOFT_TIMER_FREQUENCY 32768
>
> Are there any possible side-effects somewhere else ?
> I don't think but I may be missing some corner case.
>
> Also in contiki code you used it as
>
> #define RTIMER_ARCH_SECOND (SOFT_TIMER_FREQUENCY)
>
> Is there any reason for the parenthesis for constants ?
> Should they be added here by default also ?
>
> Is it just because it's good practice for macros in general or is there a
> reason for doing it around a constant ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iot-lab/openlab/pull/5/files#r53036521>.
>